internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -246,15 +247,19 @@ func (r *NumaflowControllerReconciler) reconcile(
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("unable to determine the target objects for the new version %s: %w", newVersion, err)
 	}
-
 	numaLogger.Debugf("found %d target objects associated with NumaflowController version %s", len(newVersionTargetObjs), newVersion)
 
+	existingClusterResources, err := r.determineExistingManagedResourceInCluster(ctx, namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to determine existing managed resources in cluster: %w", err)
+	}
```

Comment:
In one of these functions, it would be great if we can add debug level log(s) for both target objects and existing objects. I'm imagining after we apply to the tech-pi cluster, we'll check that log and make sure it looks good. And if anything were to ever go wrong with this (which I don't think it should), we'll have that log line to look back at.

Reply:
okay...but it not straight forward to log the resource as it's "array to point object" so it will print address if log directly, need to loop through all the resource then log it as value? 
BTW I have added the length log for now!!

Reply:
> need to loop through all the resource then log it as value?

I'm thinking yeah, let's do that

Reply:
Added